mmv1/third_party/terraform/services/datastream/resource_datastream_connection_profile_test.go

```diff
@@ -247,3 +272,88 @@ resource "google_datastream_connection_profile" "mysql_con_profile" {
 }
 `, context)
 }
+
+func testAccDatastreamConnectionProfile_SSHKey_Update(context map[string]interface{}, preventDestroy bool, private_key string) string {
+	context["lifecycle_block"] = ""
+	if preventDestroy {
+		context["lifecycle_block"] = `
+        lifecycle {
+            prevent_destroy = true
+        }`
+	}
+
+	context["private_key"] = private_key
+
+	return acctest.Nprintf(`
+resource "google_sql_database_instance" "instance" {
+        name             = "tf-test-my-database-instance%{random_suffix}"
+        database_version = "POSTGRES_14"
+        region           = "us-central1"
+        settings {
+            tier = "db-f1-micro"
+            ip_configuration {
+    
+            // Datastream IPs will vary by region.
+            authorized_networks {
+                value = "34.71.242.81"
+            }
+    
+            authorized_networks {
+                value = "34.72.28.29"
+            }
+    
+            authorized_networks {
+                value = "34.67.6.157"
+            }
+    
+            authorized_networks {
+                value = "34.67.234.134"
+            }
+    
+            authorized_networks {
+                value = "34.72.239.218"
+            }
+        }
+    }
+    
+        deletion_protection  = "false"
+    }
+    
+    resource "google_sql_database" "db" {
+        instance = google_sql_database_instance.instance.name
+        name     = "db"
+    }
+    
+    resource "random_password" "pwd" {
```

Comment:
It looks like the test is skipped in VCR (meaning we don't run it in PR checks) because the random_password provider is used, but I don't think you need it. You can probably pass in a random value instead, and then we can make sure the tests pass in the PR.